File: 2363-merge-similar-items/2363-merge-similar-items.swift
class Solution {
    func mergeSimilarItems(_ items1: [[Int]], _ items2: [[Int]]) -> [[Int]] {
        var someDict = [Int: Int]()
        
        for kv in items1 + items2 {
            someDict[kv[0], default: 0] += kv[1]
        }
        
        var someInts:[[Int]] = []
        
        for (key, value) in someDict {
            someInts.append([key, value])
        }
        
        return someInts.sorted(by: { $0[0] < $1[0] })
    }
}

File: 2363-merge-similar-items/2363-merge-similar-items.py
class Solution:
    def mergeSimilarItems(self, items1: List[List[int]], items2: List[List[int]]) -> List[List[int]]:
        hm = defaultdict(int)
        
        for value, weight in items1 + items2:
            hm[value] += weight
        
        return sorted([[value, hm[value]] for value in hm])